### ISSUE TITLE
3.x compose support: exception are not caught in toState()

### ISSUE THREAD

**Issue by BoD** [apollographql/apollo-kotlin#5019]
### Description

When calling toState(), exceptions should be caught and exposed via the `.exception` extension.

**Comment by BoD** [comment]
A fix is pending in #5018.

In the meantime, you can add your own extension with the same API, and call it instead of the library's:

```kotlin
@ApolloExperimental
@Composable
@OptIn(ApolloInternal::class)
fun <D : Operation.Data> ApolloCall<D>.toState(context: CoroutineContext = EmptyCoroutineContext): State<ApolloResponse<D>?> {
  val responseFlow = remember {
    toFlow()
        .catch { emit(ApolloResponse(this@toState, it as? ApolloException ?: throw it)) }
  }
  return responseFlow.collectAsState(initial = null, context = context)
}
```

**Comment by martinbonnin** [comment]
Available in `4.0.0-alpha.3`
